src/v/http/CMakeLists.txt

```diff
@@ -10,6 +10,8 @@ v_cc_library(
     Seastar::seastar
     v::bytes
     v::rpc
+  DEFINES
+    -DBOOST_ASIO_HAS_STD_INVOKE_RESULT
```

Comment:
why haven't we needed this yet? is there another change where this will be required?

Reply:
`std:: result_of`wasn't removed in libc++ 12. It has been in 13.0.0.

Reply:
got it